components/authentication/base/rh-idp/group-sync/konflux-rover-groups.yaml

```diff
@@ -0,0 +1,38 @@
+apiVersion: redhatcop.redhat.io/v1alpha1
+kind: GroupSync
+metadata:
+  name: konflux-rover-groups
+  annotations:
+    argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
+spec:
+  providers:
+    - ldap:
+        ca:
+          key: ca.crt
+          kind: Secret
+          name: mtls-ca-validators
+          namespace: group-sync-operator
+        insecure: false
+        rfc2307:
+          usersQuery:
+            baseDN: 'dc=redhat,dc=com'
+            derefAliases: never
+            scope: sub
+          groupNameAttributes:
+            - cn
+          tolerateMemberNotFoundErrors: true
+          tolerateMemberOutOfScopeErrors: true
+          groupUIDAttribute: dn
+          groupMembershipAttributes:
+            - uniqueMember
+          userNameAttributes:
+            - uid
+          groupsQuery:
+            baseDN: 'ou=adhoc,ou=managedGroups,dc=redhat,dc=com'
+            derefAliases: never
+            filter: (&(objectClass=rhatRoverGroup)(cn=konflux-*))
```

Comment:
@gbenhaim Here we have 2 options, the one implemented is to sync any groups starting with "konflux-". Another one is to explicitly mention all the groups to sync:
```
whitelist:
  - 'cn=konflux-build,ou=adhoc,ou=managedGroups,dc=redhat,dc=com'
  - 'cn=konflux-core,ou=adhoc,ou=managedGroups,dc=redhat,dc=com'
  -  ...
  - 'cn=konflux-spi,ou=adhoc,ou=managedGroups,dc=redhat,dc=com'
```
We need to decide which way we prefer.

Reply:
I think we need to mention all groups explicitly since there is nothing blocking other people form creating a groups that start with `konflux-*`

Reply:
can be done in a different pr

Reply:
If an unwanted konflux-* group is created, e.g. konflux-intruders, it will be sync but no RBACs will use it so the only harm is to have a unused k8s group.

I went with current approach because I find this cool that all you have to do for a new group is to create a rover group, then nothing else apart using it in desired RBACs. That said, I was hesitating between that as whitelisting them explicitly, thus the question.

Reply:
we can leave it with the wildcard.

Reply:
Thanks!